src/services/importService.ts

```diff
@@ -70,32 +74,57 @@ export default class ImportService implements IImportService {
      */
     public async generateAssets(v1Project: IFileInfo, v2Project: IProject): Promise<IAssetMetadata[]> {
         let originalProject: IV1Project;
-        const generatedAssetMetadata: IAssetMetadata[] = [];
-        let assetState: AssetState;
+        let generatedAssetMetadata: IAssetMetadata[] = [];
         const assetService = new AssetService(v2Project);
 
         originalProject = JSON.parse(v1Project.content as string);
 
-        for (const frameName in originalProject.frames) {
-            if (originalProject.frames.hasOwnProperty(frameName)) {
-                const frameRegions = originalProject.frames[frameName];
-                const asset = AssetService.createAssetFromFilePath(
-                    `${v1Project.file.path.replace(/[^\/]*$/, "")}${frameName}`);
-                const populatedMetadata = await assetService.getAssetMetadata(asset).then((metadata) => {
-                    assetState = originalProject.visitedFrames.indexOf(frameName) > -1 && frameRegions.length > 0
-                        ? AssetState.Tagged : (originalProject.visitedFrames.indexOf(frameName) > -1
-                        ? AssetState.Visited : AssetState.NotVisited);
-                    const taggedMetadata = this.addRegions(metadata, frameRegions);
-                    taggedMetadata.asset.state = assetState;
-                    taggedMetadata.asset.path = `file:${taggedMetadata.asset.path}`;
-                    return taggedMetadata;
-                });
-                generatedAssetMetadata.push(populatedMetadata);
-            }
+        const pathParts = v1Project.file.path.split(/[\\\/]/);
+        const fileName = pathParts[pathParts.length - 1];
+        const fileNameParts = fileName.split(".");
+
+        if (fileNameParts[1] && AssetService.getAssetType(fileNameParts[1]) === AssetType.Video) {
+            generatedAssetMetadata = await this.generateVideoAssets(v1Project, originalProject.frames, assetService);
+            v2Project.lastVisitedAssetId = generatedAssetMetadata[generatedAssetMetadata.length - 1].asset.id;
+        } else {
+            generatedAssetMetadata = await this.generateImageAssets(v1Project, originalProject.frames, assetService);
         }
         return generatedAssetMetadata;
     }
 
+    /**
+     * Generate parent asset based on V1 Project video assets
+     * @param project - V1 Project Content and File Information
+     */
+    private async createParentVideoAsset(v1Project: IFileInfo): Promise<IAsset> {
+        let parentAsset: IAsset;
+        let parentFormat: string;
+        const pathParts = v1Project.file.path.split(/[\\\/]/);
+        const fileName = pathParts[pathParts.length - 1];
+        const fileNameParts = fileName.split(".");
+        try {
+            parentFormat = fileNameParts[1];
+        } catch (e) {
+            throw e;
+        }
+
+        parentAsset = {
```

Comment:
Seems a little fragile. Any reason not to use `AssetService.createAssetFromFilePath` here?